frontend/dockerfile/parser/parser.go

```diff
@@ -178,40 +171,19 @@ func (d *directives) setEscapeToken(s string) error {
 // Parser directives must precede any builder instruction or other comments,
 // and cannot be repeated.
 func (d *directives) possibleParserDirective(line string) error {
-	if d.done {
-		return nil
-	}
-
-	match := reDirectives.FindStringSubmatch(line)
-	if len(match) == 0 {
-		d.done = true
-		return nil
-	}
-
-	k := strings.ToLower(match[1])
-	_, ok := validDirectives[k]
-	if !ok {
-		d.done = true
-		return nil
+	directive, err := d.parser.ParseLine([]byte(line))
+	if err != nil {
+		return err
 	}
-
-	if _, ok := d.seen[k]; ok {
-		return errors.Errorf("only one %s parser directive can be used", k)
+	if directive.Name == keyEscape {
+		return d.setEscapeToken(directive.Value)
 	}
-	d.seen[k] = struct{}{}
-
-	if k == "escape" {
-		return d.setEscapeToken(match[2])
-	}
-
 	return nil
 }
 
 // newDefaultDirectives returns a new directives structure with the default escapeToken token
 func newDefaultDirectives() *directives {
-	d := &directives{
-		seen: map[string]struct{}{},
-	}
+	d := &directives{}
```

Comment:
Yup, I've reworked the parser to work from a nil state easily, so that when using it outside of this package, no extra `New` methods, etc, would be required.